### PR TITLE
[Enhancement][Lake] Add publish_log_version service

### PR DESCRIPTION
**Pull request by sduzh**
This PR is part of the PR stack to support schema changes on lake
tables.
This PR aims to support finding txn log by the commit version of a
transaction.

## What type of PR is this：
- [ ] bug
- [ ] feature
- [x] enhancement
- [ ] refactor
- [ ] others

### CODE REVIEW

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

File: be/src/service/service_be/lake_service.cpp [hunk 1]
```diff
@@ -133,9 +141,43 @@ void LakeServiceImpl::publish_version(::google::protobuf::RpcController* control
...
+                                          ::starrocks::lake::PublishLogVersionResponse* response,
+                                          ::google::protobuf::Closure* done) {
+    brpc::ClosureGuard guard(done);
+    auto cntl = static_cast<brpc::Controller*>(controller);
+
+    if (request->tablet_ids_size() == 0) {
+        cntl->SetFailed("missing tablet_ids");
+        return;
+    }
+    if (request->has_txn_id() == 0) {
```
Comment (wyb) [inline be/src/service/service_be/lake_service.cpp:158]:
```suggestion
    if (!request->has_txn_id()) {
```

File: be/src/service/service_be/lake_service.cpp [hunk 2]
```diff
@@ -133,9 +141,43 @@ void LakeServiceImpl::publish_version(::google::protobuf::RpcController* control
...
+    }
+
+    auto thread_pool = _env->agent_server()->get_thread_pool(TTaskType::PUBLISH_VERSION);
+    auto context = std::make_shared<PublishLogVersionContext>(_env, request, response, request->tablet_ids_size());
+
+    for (auto tablet_id : request->tablet_ids()) {
+        auto task = std::make_shared<PublishLogVersionTask>(tablet_id, context);
+        auto st = thread_pool->submit(std::move(task));
+        if (!st.ok()) {
+            LOG(WARNING) << "Fail to submit publish version task: " << st;
```
Comment (wyb) [inline be/src/service/service_be/lake_service.cpp:174]:
```suggestion
            LOG(WARNING) << "Fail to submit publish log version task: " << st;
```

File: be/src/storage/lake/fixed_location_provider.h [hunk 3]
```diff
@@ -22,6 +22,8 @@ class FixedLocationProvider : public LocationProvider {
 
     std::string txn_log_location(int64_t tablet_id, int64_t txn_id) const override;
 
+    std::string txn_vlog_location(int64_t tablet_id, int64_t txn_id) const override;
```
Comment (wyb) [inline be/src/storage/lake/fixed_location_provider.h:25]:
```suggestion
    std::string txn_vlog_location(int64_t tablet_id, int64_t version) const override;
```

File: be/src/service/service_be/lake_service.cpp (hunk 1 above)
Comment (sduzh) [reply be/src/service/service_be/lake_service.cpp]: updated

File: be/src/service/service_be/lake_service.cpp (hunk 2 above)
Comment (sduzh) [reply be/src/service/service_be/lake_service.cpp]: updated

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by sduzh** [comment]
run starrocks_admit_test

**Review by wyb — APPROVED** [review]

**Review by gengjun-git — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**